extensions/uielement/internal.m

```diff
@@ -42,15 +42,20 @@ static bool is_window(AXUIElementRef element, NSString* role) {
     role = get_prop(element, NSAccessibilityRoleAttribute, @"");
   }
 
-  // The role attribute on a window can potentially be something
-  // other than kAXWindowRole (e.g. Emacs does not claim kAXWindowRole)
-  // so we will do the simple test first, but then also attempt to duck-type
-  // the object, to see if it has a property that any window should have
-  if([role isEqualToString: (NSString*)kAXWindowRole] ||
-     get_prop(element, NSAccessibilityMinimizedAttribute, nil)) {
-    return YES;
+  if ([role isKindOfClass:[NSString class]]) {
+      // The role attribute on a window can potentially be something
+      // other than kAXWindowRole (e.g. Emacs does not claim kAXWindowRole)
+      // so we will do the simple test first, but then also attempt to duck-type
+      // the object, to see if it has a property that any window should have
+      if([role isEqualToString: (NSString*)kAXWindowRole] ||
+         get_prop(element, NSAccessibilityMinimizedAttribute, nil)) {
+        return YES;
+      } else {
+        return NO;
+      }
   } else {
-    return NO;
+      [LuaSkin logBreadcrumb:[NSString stringWithFormat:@"%s:is_window AXRole is not a string type", watcherTag]] ;
+      return NO;
   }
 }
 
```

Comment:
I think that you also need to amend the push_element function:

Changing
``` lua
// Use the Role of the element to decide which type of object to create: window, app, or plain uielement.
// Retains a copy of the element, if necessary.
static void push_element(lua_State* L, AXUIElementRef element) {
    NSString* role = get_prop(element, NSAccessibilityRoleAttribute, @"");

    if (is_window(element, role)) {
        new_window(L, (AXUIElementRef)CFRetain(element));
    } else if ([role isEqualToString: (NSString*)kAXApplicationRole]) {
        pid_t pid;
        AXUIElementGetPid(element, &pid);
        if (!new_application(L, pid)) {
            lua_pushnil(L);
        }
    } else {
        new_uielement(L, (AXUIElementRef)CFRetain(element));
    }
}
```
to
``` lua
// Use the Role of the element to decide which type of object to create: window, app, or plain uielement.
// Retains a copy of the element, if necessary.
static void push_element(lua_State* L, AXUIElementRef element) {
    NSString* role = get_prop(element, NSAccessibilityRoleAttribute, @"");

    if (is_window(element, role)) {
        new_window(L, (AXUIElementRef)CFRetain(element));
    } else if ([role isKindOfClass:[NSString class]] && [role isEqualToString: (NSString*)kAXApplicationRole]) {
        pid_t pid;
        AXUIElementGetPid(element, &pid);
        if (!new_application(L, pid)) {
            lua_pushnil(L);
        }
    } else {
        new_uielement(L, (AXUIElementRef)CFRetain(element));
    }
}
```
(single line change)